src/mozillavpn.cpp

```diff
@@ -1805,7 +1808,16 @@ void MozillaVPN::registerNavigationBarButtons() {
       MozillaVPN::ScreenSettings, "qrc:/nebula/resources/navbar/settings.svg",
       "qrc:/nebula/resources/navbar/settings-selected.svg"));
 
-  connect(SettingsHolder::instance(), &SettingsHolder::addonSettingsChanged,
+  // A group of settings containing all the addon message settings.
+  SettingGroup messageSettingGroup =
+      SettingGroup(QString("%1/%2")
+                       .arg(Constants::ADDONS_SETTINGS_GROUP)
+                       .arg(ADDON_MESSAGE_SETTINGS_GROUP),
+                   true,  // remove when reset
+                   false  // sensitive setting
+      );
+
```

Comment:
Consider if a design like the following is better, after introducing a singleton SettingsManager:

  `SettingGroup messageSettingGroup =
      SettingGroup(QString("%1/%2")
                       .arg(Constants::ADDONS_SETTINGS_GROUP)
                       .arg(ADDON_MESSAGE_SETTINGS_GROUP),
                   true,  // remove when reset
                   false  // sensitive setting
      );

  SettingsManager::instance()->register(messageSettingGroup);`

This improves the Single Responsibility Principle, by moving the registration action out of SettingGroup.